akka-http-core/src/main/scala/akka/http/scaladsl/ClientTransport.scala

```diff
@@ -57,11 +56,15 @@ object ClientTransport {
    */
   def httpsProxy(proxyAddress: InetSocketAddress): ClientTransport = new HttpsProxyTransport(proxyAddress)
 
-  private case class HttpsProxyTransport(proxyAddress: InetSocketAddress, underlyingTransport: ClientTransport = TCP) extends ClientTransport {
+  def httpsProxy(proxyAddress: InetSocketAddress, proxyAuth: HttpCredentials): ClientTransport = new HttpsProxyTransport(proxyAddress, proxyAuth = Some(proxyAuth))
```

Comment:
Needs Scaladoc and `@ApiMayChange` marker.

The same change is needed for `javadsl.ClientTransport`.